sdk/servicebus/service-bus/src/serviceBusAtomManagementClient.ts

```diff
@@ -1423,7 +1470,10 @@ export class ServiceBusManagementClient extends ServiceClient {
       queueOrSubscriptionFields.ForwardTo ||
       queueOrSubscriptionFields.ForwardDeadLetteredMessagesTo
     ) {
-      const token = (await this.sasTokenProvider.getToken(this.endpoint))!.token;
+      const token =
+        this.tokenProvider instanceof SharedKeyCredential
+          ? this.tokenProvider.getToken(this.endpoint)!.token
+          : (await this.tokenProvider.getToken([]))!.token;
```

Comment:
I'm curious why an empty array is passed when using AAD credentials (I'm not very familiar with this part).
Do we have tests using AAD credentials for any of the operations that require this token?

Reply:
Ah, I was getting there.
Regarding the empty array, getToken would fail. I can add `AMQPConstants.aadServiceBusScope` to fix. I have some more observations to share.

I was testing this part of code last night and noticed that the token can even just be an empty string `""`, the headers don't care what the token is, the requests go through and my tests succeeded(Both with the connection string and the AAD).

I don't know the motivation behind adding this getToken here. I am considering making it an empty string `""` since it seems work and just not sure if I'm missing anything.

Reply:
Weird...we should probably reach out to the service team then if we're unclear why we're adding these headers (unless someone from .NET knows!) but that can be a follow up after this PR.

Reply:
Shared findings with @ShivangiReja, she's verifying the .NET SDK.
We may need to reach out to the service team.

Reply:
It shouldn't be empty, token is needed when we create an entity.

Reply:
Thanks Shivangi!

Reply:
Dumping my investigation here..

1. Initially, I was testing this with `updateQueue()` method which I later realized that even though the code path is hit, the service doesn't care about the `forwardTo` attribute while doing update.
Reference - https://docs.microsoft.com/en-us/rest/api/servicebus/update-queue
2. Tested out this with `createQueue`, fixed a couple of bug fixes related to trailing slashes and I now confirm that the token is required when `forwardTo` is not undefined.
AAD path is working fine for the `forwardTo` attribute now that I've fixed the issues, I'm running into errors when dealing with the connection string, looking into it now.

Reply:
Fixed!!